app/src/main/java/com/github/fribourgsdp/radio/GameSettingsActivity.kt

```diff
@@ -105,49 +105,46 @@ class GameSettingsActivity : AppCompatActivity() {
                 putExtra(GAME_NAME_KEY, nameInput.text.toString().ifEmpty { getString(R.string.default_game_name) })
                 putExtra(GAME_PLAYLIST_KEY, json.encodeToString(selectedPlaylist))
                 putExtra(GAME_NB_ROUNDS_KEY, nbRoundsInput.text.toString().ifEmpty { getString(R.string.default_game_nb_rounds) }.toInt())
-                putExtra(GAME_HINT_KEY, hintCheckBox.isChecked,)
+                putExtra(GAME_HINT_KEY, hintCheckBox.isChecked)
                 putExtra(GAME_PRIVACY_KEY, privacyCheckBox.isChecked)
                 putExtra(GAME_IS_HOST_KEY, true)
             }
             startActivity(intent)
         }
     }
+    private fun getPlaylist(name: String): Playlist {
+        return host.getPlaylists().find { playlist -> playlist.name == name }!!
+    }
 
-    private fun searchViewOnQueryBehavior() : SearchView.OnQueryTextListener {
-        return object : SearchView.OnQueryTextListener {
-            override fun onQueryTextSubmit(query: String?): Boolean {
-                // When selected, hide the possibilities
-                playlistListView.visibility = View.GONE
-
-                if (playlistsNames.contains(query) && query != null) {
-                    playlistAdapter.filter.filter(query)
-                    selectedPlaylist = getPlaylist(query)
-                    startButton.isEnabled = true
-                    errorText.visibility = View.GONE
-                } else {
-                    errorText.text = getString(R.string.playlist_error_format, query)
-                    errorText.visibility = View.VISIBLE
-                    startButton.isEnabled = false
-                }
+    inner class SearchViewOnQueryBehavior: SearchView.OnQueryTextListener {
```

Comment:
Does `inner class` make this class private? Because I don't think this should be accessed outside of `GameSettingsActivity`.

Reply:
It is indeed accessible outside so you should make this private.

Reply:
Also, as you can read in the [doc for Inner classes](https://kotlinlang.org/docs/nested-classes.html#inner-classes), with the keyword `inner`, this class can access the members of its outer class. So it's better to write :
```kotlin
private class SearchViewOnQueryBehavior: SearchView.OnQueryTextListener { …
```

Reply:
Bien vu, I'll change that.